_posts/2021/2021-03-19-the-x-factor.md

```diff
@@ -0,0 +1,59 @@
+
+The underlying cause for the above issues is the separation of teams. Each team has its own scheduled meetings, sits in its own room, and has its own inside jokes (hopefully not just lame movie puns). Changing an inappropriate API design or usage involves walking over to the other room, and finding someone who then has to switch context, understand the problem, and maybe even create a ticket for it. Since programmers are lazy, they'll start taking shortcuts to avoid this overhead, at the cost of inefficient and obscure API usage. It's easy to complain about "them" not doing "the right thing" for "us", or "them" using "our" API wrong, when "they" are in another team.
+
+## xXx functional – Triple X
+
+When I joined the new cross functional team, it was a real cultural shock. I had to learn so many new terms, processes, and tools, that had previously been none of my business, or hidden behind JSON APIs. The complexity and various dependencies in the cloud based microservice setup used by our backend was overwhelming. My goal was to understand, be able to review, and contribute to the parts of the backend which my team decided to focus on. From the other end, my two backend colleagues, who wanted to review my frontend code and start contributing to the frontend projects, were overwhelmed by the complex code structure and the many coding and formatting conventions enforced there. On top of that, we were used to different processes: Scrum vs. Kanban, dedicated vs. integrated QA, or the [localization workflow](../../language-and-localization/how-to-cope-with-a-localization-tool-in-action/), which was new to the backend developers.
+
+But because we were all in one team, and all in the same situation, there was always someone available and happy to educate the others about any of those topics. Even though Corona has made me spend no more than one (!) day in the office ever since I've joined the team, the inhibition threshold to have a short video call about any open question has always been very low. Contributing code in a different programming language to "the other end" (in my case Java to the backend) gave me worthful insights, broadened my own horizon, and helped me gain another perspective on the overall software architecture. Likewise, my colleagues have made similar experiences contributing to the frontend code.
```

Comment:
maybe you could add some factors that made this possible? team size? investments in team building? open attitudes? (after all, only people joined that were open for a change in the first place)...